src/aggregate/aggregate_exec.c

```diff
@@ -259,10 +260,15 @@ static int buildRequest(RedisModuleCtx *ctx, RedisModuleString **argv, int argc,
     goto done;
   }
 
+  if (IsProfile(*r)) {
+    parseClock = clock();
+    (*r)->parseTime = parseClock - (*r)->initClock;
+  }
+
   rc = AREQ_BuildPipeline(*r, 0, status);
 
   if (IsProfile(*r)) {
-    (*r)->parseTime = clock() - (*r)->initClock;
```

Comment:
Seems worthwhile calling `clock()` only once

Reply:
Looked again, my mistake 😊 - `clock()` is not called more times  than it should.